tests/test_model.py

```diff
@@ -1833,8 +1823,12 @@ def test_open_access_loan(self):
         eq_(loan, loan2)
         eq_(False, was_new)
 
+
     def test_work(self):
         """Test the attribute that finds the Work for a Loan or Hold."""
+        # TODO:  determine if using pool.presentation_edition is correct for lending.
+        # Because, while the old pool.edition was tied to something you can lend, the presentation_edition 
+        # is not representing a lendable object, really.  
```

Comment:
I think I see what you're saying. 

This test is testing the property `LoanAndHoldMixin.work`. If everything is working properly, the `LicensePool` has a `.work` set and that's the end of it.

But there might be a problem such that `LicensePool.work` is not set. In this case (which shouldn't happen) we might still be able to find the appropriate `.work` because of some redundancy in the database. Specifically, the LicensePool's `.presentation_edition` might be associated with some `.work`. This will always be the `Work` that `LicensePool.work` _should_ have been set to, so we can check that and fall back to it.

Currently we can generally rely on `LicensePool.presentation_edition` having an associated `.work`, even if `Edition.is_primary_for_work` is not set. However that might change in the future because we might be getting rid of that redundancy. If we get rid of `Edition.is_primary_for_work`, then an `Edition` will only have a `.work` if it is the primary for its work. 

Most of the time you'll still be able to get from `LicensePool` to `Edition` to `Work`, because most of the time a `Work` will only have one `LicensePool`. But when there are multiple `LicensePool`s, it's always possible that some other `LicensePool` is the one that provides the `.presentation_edition` for the work.

Since this is fallback code and there will eventually be situations where the fallback doesn't work, I'm okay with taking the code out of `LoanAndHoldMixin.work` and taking out this part of the test.